server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -141,9 +269,32 @@ public static IndicesOptions fromOptions(boolean ignoreUnavailable, boolean allo
 
     public static IndicesOptions fromOptions(boolean ignoreUnavailable, boolean allowNoIndices, boolean expandToOpenIndices,
```

Comment:
Would replacing the usage of this method with the new constructor be preferred?

```
new IndicesOptions(Sets.newHashSet(Option.IGNORE_UNAVAILABLE, Option.ALLOW_NO_INDICES), Collections.emptySet());
```

Now that we have enum sets we can leverage to create random instances in tests, do you see a need for this method? I understand these are widely used in other places and would be a slight pain to refactor, but we will most likely need to either refactor, or add extra method signatures for any new flag we would want to implement.

Reply:
I think that we should aim at removing these methods, they are trappy as it is never clear which one to use and some include default values for some of the flags. The main problem with this is that internal flags (e.g. can an API work against closed indices, or work against aliases that point to multiple indices) and flags that can be set via our API (e.g. ignore_unavailable, expand_wildcards, allow_no_indices) are mixed up together. I would do this as a follow-up though.

Reply:
I tried to keep this change as small as possible, figuring that we could always change the methods once the implementation was decided on

Reply:
👍 @dakrone. I'm happy to follow-up with a PR to refactor this